pkg/tsdb/cloudwatch/response_parser.go

```diff
@@ -48,10 +48,16 @@ func (e *cloudWatchExecutor) parseResponse(startTime time.Time, endTime time.Tim
 func aggregateResponse(getMetricDataOutputs []*cloudwatch.GetMetricDataOutput) map[string]queryRowResponse {
 	responseByID := make(map[string]queryRowResponse)
 	for _, gmdo := range getMetricDataOutputs {
-		requestExceededMaxLimit := false
+		errorCodes := map[string]bool{
```

Comment:
It's better to declare and set this variable outside of the loop. That way we'll catch the scenario where only the first GetMetricDataOutput returns the error. I'm not sure that's a valid scenario and I know this was broken before too, but better safe than sorry so please change this. Would be great if you could add another test case for this too. You can copy multiple-outputs.json, modify it and name it multiple-outputs2.json.

Reply:
Not sure If I am understanding the issue here. If am understanding correctly, the current approach also catches the scenario you are describing. By moving the declaration and setting of `errorCodes` to outside the loop, the `errorCodes` will be shared between all `GetMetricDataOutput` in the loop. Let me know if I am thinking wrong.

Reply:
If you keep setting the variable to false in the loop, it will be reset to false for every item. We want to act upon the occurrence of an error in any of the `GetMetricDataOutput`, not matter where in the array it is. You can try this in your unit tests by moving the errors to the first `GetMetricDataOutput`. See the json below.   

<details>
 <summary>multiple-json.output
 </summary>

```json
[
  {
    "Messages": [
      { "Code": "", "Value": null },
      { "Code": "MaxMetricsExceeded", "Value": null },
      { "Code": "MaxQueryTimeRangeExceeded", "Value": null },
      { "Code": "MaxQueryResultsExceeded", "Value": null },
      { "Code": "MaxMatchingResultsExceeded", "Value": null }
    ],
    "MetricDataResults": [
      {
        "Id": "a",
        "Label": "label1",
        "Messages": null,
        "StatusCode": "Complete",
        "Timestamps": [
          "2021-01-15T19:44:00Z",
          "2021-01-15T19:59:00Z",
          "2021-01-15T20:14:00Z",
          "2021-01-15T20:29:00Z",
          "2021-01-15T20:44:00Z"
        ],
        "Values": [
          0.1333395078879982,
          0.244268469636633,
          0.15574387947267768,
          0.14447563659125626,
          0.15519743138527173
        ]
      },
      {
        "Id": "a",
        "Label": "label2",
        "Messages": null,
        "StatusCode": "Complete",
        "Timestamps": [
          "2021-01-15T19:44:00Z"
        ],
        "Values": [
          0.1333395078879982
        ]
      },
      {
        "Id": "b",
        "Label": "label2",
        "Messages": null,
        "StatusCode": "Complete",
        "Timestamps": [
          "2021-01-15T19:44:00Z"
        ],
        "Values": [
          0.1333395078879982
        ]
      }
    ],
    "NextToken": null
  },
  {
    "Messages": null,
    "MetricDataResults": [
      {
        "Id": "a",
        "Label": "label1",
        "Messages": null,
        "StatusCode": "Complete",
        "Timestamps": [
          "2021-01-15T19:44:00Z",
          "2021-01-15T19:59:00Z",
          "2021-01-15T20:14:00Z",
          "2021-01-15T20:29:00Z",
          "2021-01-15T20:44:00Z"
        ],
        "Values": [
          0.1333395078879982,
          0.244268469636633,
          0.15574387947267768,
          0.14447563659125626,
          0.15519743138527173
        ]
      },
      {
        "Id": "b",
        "Label": "label2",
        "Messages": [{
          "Code": "ArithmeticError",
          "Value": "One or more data-points have been dropped due to non-numeric values (NaN, -Infinite, +Infinite)"
        }],
        "StatusCode": "Partial",
        "Timestamps": [
          "2021-01-15T19:44:00Z"
        ],
        "Values": [
          0.1333395078879982
        ]
      }
    ],
    "NextToken": null
  }
]

```

</details>

Reply:
I take back what I said, I was make a programmatic mistake so did not see that error. Anyway, here's the commit with the changes given your feedback.